### PR TITLE
Remove UseIdentity and IdentityVersion from libuplink volatile config

### PR DESCRIPTION
**Pull request by kaloyan-raev**
What: 

- Removes `UseIdentity` and `IdentityVersion` from libuplink `Config.Volatile`. We shouldn't have aby use case for these two anymore.
- storj-sim does not generate identity for gateway anymore
- Gateway properly closes the libuplink project when verifying the satellite config

Why:

These changes address additional comments left in #1920 and #1921.

Thanks for submitting a PR!

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: cmd/gateway/main.go [hunk 1]
```diff
@@ -165,6 +159,21 @@ func generateKey() (key string, err error) {
...
 
+func checkCfg(ctx context.Context) (err error) {
+	proj, err := runCfg.openProject(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() { err = errs.Combine(err, proj.Close()) }()
+
+	_, err = proj.ListBuckets(ctx, &storj.BucketListOptions{Direction: storj.After})
+	if err != nil {
```
Comment (mniewrzal) [inline cmd/gateway/main.go]: nit: you can do just
```
_, err = proj.ListBuckets
return err
```

Comment (kaloyan-raev) [reply cmd/gateway/main.go]: You are absolutely right. Fixed.

**Comment by thepaul** [comment]
I'm just seeing this now. Can we consider reverting the change? I think it needs to be possible to supply your own identity. Definitely not required, but it should be an option.

I want to be able to specify things like "uplink_identity=SomeIDHash" in macaroons. That might end up being the best way to implement access control restrictions that depend on authn of the requester. It would require some more plumbing to be user-friendly (maybe making some way of telling the satellite to associate email address A with identity B, and a way of querying that), but it seems a lot better than configuring an uplink to send a username and password or something.

Also, there might be cases where a user wants uplink setup to be as fast as possible. I imagine they'd want to pre-generate their identity and use it in the config.

**Comment by ifraixedes** [comment]
It sounds an interesting use case.
I would say, rather than reverting it, adding back the identity outside of volatile making optional and add to the docs when they are good to be set.

**Comment by jtolds** [comment]
> I want to be able to specify things like "uplink_identity=SomeIDHash" in macaroons.

oh, neat! that's a great idea.

**Comment by jtolds** [comment]
that sounds useful enough that we would want to promote it out of volatile anyway? should we do a PR that adds identity as a first class setup option?

**Comment by thepaul** [comment]
Sounds ok to me.

**Comment by ifraixedes** [comment]
It has been registered under the issue [V3-1785](https://storjlabs.atlassian.net/browse/V3-1785?utm_term=&utm_source=jira-slack&utm_medium=referral-external&atlOrigin=eyJpIjoiNjZkYjE0NTE0YzM1NDBlNDkyODdiN2MwZDk1ZDcyNmEiLCJwIjoiamlyYS1zbGFjay1pbnQifQ)

**Pull request merged**